include/odp/api/spec/traffic_mngr.h

```diff
@@ -549,6 +573,41 @@ void odp_tm_egress_init(odp_tm_egress_t *egress);
 int odp_tm_capabilities(odp_tm_capabilities_t capabilities[],
 			uint32_t              capabilities_size);
 
+/** Query All TM Capabilities specific to an egress
+ *
+ * The odp_tm_egress_capabilities() function can be used to obtain the complete
+ * set of TM limits supported by this implementation for a given egress.
+ * The reason that this  returns a SET of capabilities and not just one, is
+ * because it is expected  that many HW based implementations may have one set
+ * of limits for the HW and also support a SW TM implementation with a
+ * (presumably larger) different set of limits.  There are also cases where
+ * there could be more than SW implementation (one supporting say tens of
+ * thousands of tm_queues and a variant supporting tens of millions of
+ * tm_queues). This is needed as platforms might have different constraints
+ * for different egress type and there is no way to know them using
+ * odp_tm_capabilities() provided capabilities. The caller passes in an array
+ * of odp_tm_capabilities_t records  and the number of such records.  Then the
+ * first N of these records will be  filled in by the implementation and the
+ * number N will be returned.  In the event that N is larger than the
+ * capabilities_size, N will still be returned, but only capabilities_size
+ * records will be filled in.
```

Comment:
If there are more than one TM implementation for a given egress and they have different capabilities, how is the one used selected? I do not think the requirement struct currently has a field for every capability so can there be a problem that at TM creation the implementation all implementation alternatives for a given egress have sufficient capabilities and the implementation just picks one of them but then later it turns out that it should have picked an other one?

It would improve readability if the description was not all in single paragraph.